api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -297,4 +297,18 @@ default AppendFiles newFastAppend() {
    * Returns a {@link LocationProvider} to provide locations for new data files.
    */
   LocationProvider locationProvider();
+
+  /**
+   * Returns the current refs for the table
+   *
+   * @return the current refs for the table
+   */
+  Map<String, SnapshotRef> refs();
```

Comment:
Do we want to expose `SnapshotRef` in the `Table` API? Why not return Snapshot for ref names?

Reply:
I was thinking it made sense to expose Refs in the Table API because refs are maintained per table and any operation which leverages the table API can easily access them.

 I think refs() is helpful mostly as a convenience method to list all the refs in a table (similar being able to list the table properties, or the schemas/partition specs etc).

For the cases where a user of the table API knows what ref they are looking for then the ref(ref) becomes helpful. If we don't return Snapshot for a given ref name, then a caller has to do 

`table.snapshot(table.refs(name).snashotId())`

and they have to also take care of the null check in case a ref with name does not exist.

So my conclusion is the following:

1.) Expose the SnapshotRef in the Table API should be fine, because refs are maintained at the table level and an API expose to them either as a collection or a convenience method to look them up by name fits the model, and could be used.

2.) It will be common to want the snapshot for a given ref, and it also makes sense to have an API for returning a Snapshot for a given ref name.

Alternative:

The tradeoff of the above is the API is less minimal. For the purpose of the table scan we will ultimately need the snapshot for a given ref. So if we want to start minimal, what we can do is add the 

`Snapshot snapshot(String ref)` signature and only when the refs(), ref(String name) are truly needed we add those. 

 @rdblue @jackye1995 @singhpk234 Let me know your thoughts!

Reply:
For now, I'd probably add the `snapshot(String name)` method and put off adding methods to `Table` until we know that we will definitely need them.

Reply:
Actually, one thing I forgot was for SerializableTable, we'll need to pass the refs through https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/SerializableTable.java#L83 so we'll need Map<String, SnapshotRef> refs() to implement the contract of snapshot(String name) in SerializableTable.

So then we need both refs() and snapshot(String name). We can leave off the ref(String name) for now, until we know that convenience method is really useful.